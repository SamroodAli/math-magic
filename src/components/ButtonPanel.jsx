import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import './style/ButtonPanel.css';

const ButtonPanel = ({ onOperationClick, onNumberClick }) => (
  <div className="panel">
    <div>
      <Button color="black" onClick={onOperationClick} name="AC" data-testid="button" />
      <Button color="black" onClick={onOperationClick} name="+/-" data-testid="button" />
      <Button color="black" onClick={onOperationClick} name="%" data-testid="button" />
      <Button color="black" onClick={onOperationClick} name="รท" data-testid="button" />
    </div>
    <div>
      <Button color="black" onClick={onNumberClick} name="7" data-testid="button" />
      <Button color="black" onClick={onNumberClick} name="8" data-testid="button" />
      <Button color="black" onClick={onNumberClick} name="9" data-testid="button" />
      <Button color="black" onClick={onOperationClick} name="x" data-testid="button" />
    </div>
    <div>
      <Button color="black" onClick={onNumberClick} name="4" data-testid="button" />
      <Button color="black" onClick={onNumberClick} name="5" data-testid="button" />
      <Button color="black" onClick={onNumberClick} name="6" data-testid="button" />
      <Button color="black" onClick={onOperationClick} name="-" data-testid="button" />
    </div>
    <div>
      <Button color="black" onClick={onNumberClick} name="1" data-testid="button" />
      <Button color="black" onClick={onNumberClick} name="2" data-testid="button" />
      <Button color="black" onClick={onNumberClick} name="3" data-testid="button" />
      <Button color="black" onClick={onOperationClick} name="+" data-testid="button" />
    </div>
    <div>
      <Button color="black" onClick={onNumberClick} name="0" data-testid="button" />
      <Button color="black" onClick={onNumberClick} name="." data-testid="button" />
      <Button color="black" onClick={onOperationClick} name="=" data-testid="button" />
    </div>
  </div>
);

ButtonPanel.propTypes = {
  onOperationClick: PropTypes.func.isRequired,
  onNumberClick: PropTypes.func.isRequired,
};

export default ButtonPanel;
