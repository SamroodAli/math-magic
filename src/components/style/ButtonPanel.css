.panel div{
  height:100px;
  display:flex;
  flex-wrap: nowrap;
}
