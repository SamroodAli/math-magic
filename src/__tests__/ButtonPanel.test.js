import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';

test('Button panel displays all calculator buttons', () => {
  const onOperationClick = jest.fn();
  const onNumberClick = jest.fn();
  const tree = renderer
    .create(
      <ButtonPanel
        onOperationClick={onOperationClick}
        onNumberClick={onNumberClick}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button panel returns panel of 19 buttons', async () => {
  const onOperationClick = jest.fn();
  const onNumberClick = jest.fn();
  const buttonPanel = render(
    <ButtonPanel
      onOperationClick={onOperationClick}
      onNumberClick={onNumberClick}
    />,
  );
  const buttons = await buttonPanel.findAllByTestId('button');
  expect(buttons.length).toEqual(19);
});

test('Button panel should return button type react nodes', async () => {
  const onOperationClick = jest.fn();
  const onNumberClick = jest.fn();
  const buttonPanel = render(
    <ButtonPanel
      onOperationClick={onOperationClick}
      onNumberClick={onNumberClick}
    />,
  );
  const buttons = await buttonPanel.findAllByTestId('button');
  for (let i = 0; i < buttons.length; i += 1) {
    expect(buttons[i].type).toEqual('button');
  }
});
