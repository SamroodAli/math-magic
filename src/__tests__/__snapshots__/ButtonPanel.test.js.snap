// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button panel displays all calculator buttons 1`] = `
<div
  className="panel"
>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      x
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
  </div>
  <div>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "50%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#E0E0E0",
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
